File: app/src/main/java/djordjeh/architecture/mvp/ToDoApplication.java
package djordjeh.architecture.mvp;

import dagger.android.AndroidInjector;
import dagger.android.support.DaggerApplication;
import djordjeh.architecture.mvp.dagger.DaggerApplicationComponent;

public class ToDoApplication extends DaggerApplication {
    @Override
    protected AndroidInjector<? extends DaggerApplication> applicationInjector() {
        return DaggerApplicationComponent.builder().application(this).build();
    }
}

File: app/src/main/java/djordjeh/architecture/mvp/ToDoApplication.kt
package djordjeh.architecture.mvp

import dagger.android.AndroidInjector
import dagger.android.support.DaggerApplication
import djordjeh.architecture.mvp.dagger.DaggerApplicationComponent

class ToDoApplication : DaggerApplication() {
    override fun applicationInjector(): AndroidInjector<out DaggerApplication> {
        return DaggerApplicationComponent.builder().application(this).build()
    }
}
